src/coreclr/jit/compiler.cpp

```diff
@@ -3594,6 +3594,31 @@ bool Compiler::compStressCompileHelper(compStressArea stressArea, unsigned weigh
     return (hash < weight);
 }
 
+//------------------------------------------------------------------------
+// compPromoteLessStructs: helper to determine if the local
+//   should not be promoted under a stress mode.
+//
+// Arguments:
+//   lclNum - local number to test
+//
+// Returns:
+//   true if this local should not be promoted.
+//
+// Notes:
+//   Reject ~50% of the potential promotions if STRESS_PROMOTE_LESS_STRUCTS is active.
```

Comment:
So, the stress mode will kick in for 50% of functions, and 50% of the structs in those methods that otherwise would have been promoted will not be.

Reply:
Yes, this is an exact description.